front/app/api/initiatives/useInitiativeBySlug.ts

```diff
@@ -0,0 +1,20 @@
+import { useQuery } from '@tanstack/react-query';
+import { CLErrors } from 'typings';
+import fetcher from 'utils/cl-react-query/fetcher';
+import initiativesKeys from './keys';
+import { IInitiative, InitiativesKeys } from './types';
+
+const fetchInitiativeBySlug = (initiativeSlug: string) =>
+  fetcher<IInitiative>({
+    path: `/initiatives/by_slug/${initiativeSlug}`,
+    action: 'get',
+  });
+
+const useInitiativeBySlug = (initiativeSlug: string) => {
+  return useQuery<IInitiative, CLErrors, IInitiative, InitiativesKeys>({
```

Comment:
Should we have a similar enabled logic here or is this always used with a slug?

Reply:
The use cases i saw had it but it is probably safer to add after the experience with the image one. I'll make the change